### DISCUSSION TITLE
Suggest flow for update pass

### DISCUSSION THREAD

**Discussion by prashant03Dev** [alexandercerutti/passkit-generator#201] [Q&A]
@alexandercerutti 

As I followed the link that you shared,but I'm having some doubts related to update pass

1) How would we know when to update the pass and to send the notification on ios device.
2) As in doc mentioned serial number,is it generated on server side randomly like uuid?
3) Suggest me how will i get the deviceLibraryIdentifier,passTypeIdentifier & Serial number to send in below api
    api: POST https://yourpasshost.example.com/v1/devices/{deviceLibraryIdentifier}/registrations/{passTypeIdentifier}/{serialNumber}

@alexandercerutti Please help me

**Comment by alexandercerutti** [comment]
@prashant03Dev 

1. When you have changes to your data, you issue a notification that will trigger the chain of calls that will bring your update endpoint to get called

2. I think you should save it, in order to match the update

3. The parameters you are talking about are called, indeed, parameters. Different frameworks, have a different way to match them inside the request URL they listen to. If you listen to a specific endpoint, the framework you are using should provide them to you in an object, probably in the request. You should definitely check out the documentation related to your framework.

Please consider donating to this project as I'm helping you completely for free

**Reply by prashant03Dev** [reply]
@alexandercerutti 

As My current approach as per the apple developer documentation

When user hits the add to wallet button,request to server fires,As on backend we check if user has the assigned pass of specific type or not,If not we create pass and record its entry into our db,and re-generate the pass,

@alexandercerutti Suggest me if i am wrong,As my question is when everytime we re-assigned the pkpass file and ios device adds that to apple wallet,Will it be there duplicacy,if yes then how can we deal with this

**Reply by alexandercerutti** [reply]
And how do you check if the user has already added the pass?

Duplicates are not allowed in Apple Wallet, as the same pass is identified by the `serialNumber`, `passTypeIdentifier` and I think the `teamIdentifier`.

When you add a pass with all the three same data, PKPassViewController is not even presented.

I think that if you are generating the serialNumber completely random, you should definitely be able to generate it in a replicable way, in order to associate it with your user or activity or transaction or whatever, imho.
